core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -596,106 +812,39 @@ impl RocksDBStore {
         )
         .unwrap();
 
-        RocksDBStore {
+        StateManagerDatabase {
             config: DatabaseFlags {
                 enable_local_transaction_execution_index: false,
                 enable_account_change_index: false,
             },
-            db,
+            rocks: ReadonlyRocks {
+                wrapped: DirectRocks { db },
+            },
         }
     }
 
     pub fn try_catchup_with_primary(&self) {
```

Comment:
since we're going the traits / type safety route would it make sense to introduce `SecondaryRocks` and move it there? not that it matters much, but just for consistency

Reply:
good suggestion; I also got rid of the non-sensical `ReadonlyRocks` struct 👍